test/externalTests/crafting.js

```diff
@@ -54,14 +54,31 @@ module.exports = () => async (bot) => {
   bot.chat('/give @p stick 21')
   await once(bot.inventory, 'updateSlot')
   const craftingTable = bot.findBlock({ matching: blocksByName.crafting_table.id })
-  await bot.craft(bot.recipesFor(itemsByName.ladder.id, null, null, true)[0], 9, craftingTable)
-  if (bot.inventory.count(itemsByName.ladder.id, null) !== 9) { throw new Error('CraftLadder: count mismatch') }
 
-  bot.chat('/give @p milk_bucket 6')
+  const recipe = bot.recipesFor(itemsByName.ladder.id, null, null, true)[0]
+  await bot.craft(recipe, recipe.result.count * 3, craftingTable)
+  await bot.waitForTicks(5)
```

Comment:
I don't like the implication of this, what `bot.craft` returns the inventory transactions should be complete

Reply:
Me neither, only happens in 1.17+. Needs more investigation :/